### PR TITLE
Code Cleanup/Deprecation fixes: Class#newInstance, underscore identifier, better temp files, avoid toUpper/LowerCase and more

### PR DESCRIPTION
**Pull request by mbien**
Contains project wide cleanups/refactorings/deprecation fixes:

 - Prefer `Files#createTempFile` over `File#createTempFile`.
   - `Files.createTempFile` uses more restrictive file permissions if possible.
 - Prefer `String.equalsIgnoreCase()` over `toUpper/LowerCase` patterns
   - `String.toUpper/LowerCase()` without setting `Locale.ROOT` can lead to unintended results on certain locales.
   - followup to #5219
 - Replace single underscore identifiers with regular names.  `_` is a reserved keyword since java 8
 - Fix `Class#newInstance` deprecation warnings  
   - `c.getDeclaredConstructor().newInstance()` can be used instead
   - simplified exception handling in related code
 - Prefer `list.sort(comp)` over `Collections.sort(list, comp)`
   - more fluent / one import less in many cases
   - (there is no sort method without `Comparator`, we could in theory map `Collections.sort(list)` to `list.sort(null)`, but I left those cases out since this might look confusing)
 - [More efficient](https://shipilev.net/blog/2016/arrays-wisdom-ancients/#_conclusion) `col.toArray()`
   - zeroing step is skipped with `col.toArray(new T[0])` or `(T::new)`
   - easier to optimize for the JVM
   - shorter/easier to read
   - large commit which is split into 6 parts along cluster boundaries

All changes were made using either jackpot scripts or built-in NetBeans code transformations. The `Class#newInstance` change needed some adjustments per hand, since due to slightly different exception handling.

I looked through the patch files twice, except the `toArray()` change which I only looked through once due to its size. I only had to tweak minor formatting anomalies here and there.

### CODE REVIEW

**Comment by mbien** [comment]
@junichi11 @troizet want to look through the PHP commit? (or any other commits you are interested in)

File: php/php.dbgp/src/org/netbeans/modules/php/dbgp/models/ThreadsModel.java
Comment (junichi11) [inline php/php.dbgp/src/org/netbeans/modules/php/dbgp/models/ThreadsModel.java:128]: Remove whitespaces?

**Comment by junichi11** [comment]
We have the [PHP feature branch](https://github.com/apache/netbeans/tree/php-nb22-features) for NB22. So, please merge this after we merge the PHP feature branch. (I'm going to merge it into the master after NB21 is released.) Thanks.

cc: @tmysik

**Comment by mbien** [comment]
@junichi11 yes saw it already, I won't merge before some of the larger pending PRs are integrated.

**Comment by tmysik** [comment]
@mbien 

Thanks a lot for doing this, Michael!

File: php/php.dbgp/src/org/netbeans/modules/php/dbgp/models/ThreadsModel.java
Comment (mbien) [reply php/php.dbgp/src/org/netbeans/modules/php/dbgp/models/ThreadsModel.java:128]: done locally, will be in next update

**Comment by mbien** [comment]
I think a good way of reviewing cleanups like this is to let github generate a diff patch for a commit, e.g:
https://github.com/apache/netbeans/commit/32a42b5994652483cc5bd9a4713a6941d175ae4b.diff
unfortunately the browser doesn't show this in color, so I download it and open it in NB. This makes scanning through it really easy and fast.

Also for the reviewers:  please mention what parts you reviewed in the approve comment so that I get an idea of what is reviewed and what is not

**Review by matthiasblaesing — APPROVED** [review]
I eyeballed samples from all commits. The changes looks sane to me and nothing out of the ordinary jumped out. Thank you.

File: profiler/profiler.api/src/org/netbeans/modules/profiler/api/ProjectUtilities.java
Comment (junichi11) [inline profiler/profiler.api/src/org/netbeans/modules/profiler/api/ProjectUtilities.java:139]: Can't use `compareToIgnoreCase`?

File: profiler/profiler.ppoints/src/org/netbeans/modules/profiler/ppoints/LoadGenProfilingPoint.java
Comment (junichi11) [inline profiler/profiler.ppoints/src/org/netbeans/modules/profiler/ppoints/LoadGenProfilingPoint.java:219]: Remove whitespaces?

File: java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/AbstractObjectVariable.java
Comment (junichi11) [inline java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/AbstractObjectVariable.java:1025]: Move `);` to the above line

File: java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/AbstractObjectVariable.java
Comment (junichi11) [inline java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/AbstractObjectVariable.java:1028]: Move this the above line?

File: java/form/src/org/netbeans/modules/form/palette/PaletteUtils.java
Comment (junichi11) [inline java/form/src/org/netbeans/modules/form/palette/PaletteUtils.java:408]: Remove a whitespace?

File: java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/LibrariesNode.java
Comment (junichi11) [inline java/java.api.common/src/org/netbeans/modules/java/api/common/project/ui/LibrariesNode.java:1362]: Remove whitespaces?

**Review by junichi11 — APPROVED** [review]
Looks good to me. I've glanced through each commit. Thank you for your work!

File: profiler/profiler.api/src/org/netbeans/modules/profiler/api/ProjectUtilities.java
Comment (mbien) [reply profiler/profiler.api/src/org/netbeans/modules/profiler/api/ProjectUtilities.java:139]: oh interesting, I didn't know that this method existed. I did only transform to `equalsIgnoreCase()`

I might do this at a later point since this PR is already large enough.
But i am going to add this to my hints collection for sure https://github.com/mbien/jackpot-inspections
thanks!

Comment (mbien) [reply profiler/profiler.api/src/org/netbeans/modules/profiler/api/ProjectUtilities.java:139]: looking closer at the method it likely has unintended side effects since it sorts the array in-place, and then it returns a copy via `toArray()` (which makes no sense).

other implementations do copy first, sort it and return the copy:
https://github.com/search?q=repo%3Aapache%2Fnetbeans%20getSortedProjects&type=code

should I change it to:
```java
    public static Provider[] getSortedProjects(Provider[] projects) {
        Provider[] sorted = Arrays.copyOf(projects, projects.length);
        Arrays.sort(sorted, (p1, p2) -> getDisplayName(p1).compareToIgnoreCase(getDisplayName(p2)));
        return sorted;
    }
```
@matthiasblaesing ?

Comment (mbien) [reply profiler/profiler.api/src/org/netbeans/modules/profiler/api/ProjectUtilities.java]: added it as extra commit

**Comment by mbien** [comment]
- the two largest PRs https://github.com/apache/netbeans/pull/6960 and https://github.com/apache/netbeans/pull/7056 are both merged
 - rebased to latest master (with no conflict which is a small miracle)

planning to merge once green again

the other PRs should do fine hopefully, I don't expect many conflicts there:
<details>

```
gh pr list --limit 30 --json number,title,changedFiles --template '{{range .}}{{tablerow .number .changedFiles .title}}{{end}}'                                                                                  7089  2     README and CHANGELOG update                                                                                                                                                                                                    
7088  1     Java Code Templates for records and sealed types
7087  11    maven indexing: update lucene from 9.9.1 to 9.10.0.
7082  3     Integrate hints preferences option in the vscode extension
7081  4     Add open containing folder action.
7080  13    Fix detection of support for generation session beans from entity classes and support jakarta package names
7077  3     Fix: Settings persistence does not work for inner-class hints
7076  9     Replace usage of package `org.codehaus.plexus.util.IOUtil` with Java native methods.
7075  1     Escape java hint fix String for html and fill in multi variables.
7070  10    New file created by copy/paste in git repo ignores commit exclusion flag
7063  9     Improve priming from LSP client
7062  26    Cleanup HashMap raw type warnings..
7055  1     Adding support for Wildfly 31
7053  1687  Code Cleanup/Deprecation fixes: Class#newInstance, underscore identifier, better temp files, avoid toUpper/LowerCase and more
7034  14    Adjusting some license texts.
7032  8     Add support for GlassFish 7.0.12
7030  3     Fixed concurrency issues in GradleDaemonExecutor
7024  15    Gradle Project shall use the Java from the tooling not runtime.
7020  38    Final changes for JakartaEE adjustments
7012  1     Improved Toolbar overflow impl.
7004  9     Update guava from 32.1.2-jre to 33.0.0-jre
6999  2     JDK 21 support to Payara Platform tools v6.x in Apache NetBeans IDE
6988  223   Add support for Jakarta EE 11
6985  4     Enable code completion in JSF and JSP pages in Jakarta based projects
6984  15    Enable Servlet/Filter/Listener generation for Jakarta based projects
6892  28    Embedded Chromium Brower into NetBeans
6866  5     Allow user to manipulate Truffle breakpoints
6844  71    [Rust] Rust workspaces support.
6834  11    Added go to test ability for test/tested class in lsp server
6829  12    Support for multiple NetBeans Gradle Tooling
```
</details>

thanks for review!

**Pull request merged**
